core/src/main/java/bisq/core/api/CoreContext.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.api;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static java.lang.Thread.currentThread;
+
+@Singleton
+@Slf4j
+class CoreContext {
+
+    @Inject
+    public CoreContext() {
+    }
+
+    public boolean isApiUser() {
+        String threadName = currentThread().getName();
+        return threadName.equals("BisqDaemonMain") || threadName.contains("grpc");
```

Comment:
I find this ugly. There must be some easier and not quite as ugly way to do this.

Reply:
Resolved by commit 0e779a4